wiki/DEFINITION_OF_DONE_CRITERIA.md

```diff
@@ -0,0 +1,78 @@
+- Array inputs/outputs should also mention their expected size/dimension.
+- If there are references to the underlying algorithm, they must be cited too.
+
+
+### Python
+
+#### Design
+
+- Python class is as "near drop-in replacement" for Scikit-learn (or relevant industry standard) API as possible. This means parameters have the same names as Scikit-learn, and where differences exist, they are clearly documented in docstrings.
+- It is recommended to open an initial PR with the API design if there are going to be significant differences with reference APIs, or lack of a reference API, to have a discussion about it. 
+- Python class is pickleable and a test has been added to `cuml/test/test_pickle.py`
+- APIs use `input_to_cuml_array` to accept flexible inputs and check their datatypes and use `cumlArray.to_output()` to return configurable outputs.
+- Any internal parameters or array-based instance variables use `CumlArray`
+
+#### Testing 
+
+- Pytests for wrapper functionality against Scikit-learn using relevant datasets
+- Stress tests against reasonable inputs (e.g short-wide, tall-narrow, different numerical precision)
+- Pytests for pickle capability
+- Pytests to evaluate correctness against Scikit-learn on a variety of datasets
+- Add algorithm to benchmarks package in `python/cuml/benchmarks/algorithms.py` and benchmarks notebook in `python/cuml/notebooks/tools/cuml_benchmarks.ipynb`
+
+#### Documentation
+
+- Complete and comprehensive Pydoc strings explaining public API, restrictions, a usage example, and gotchas. This should be in [Numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) format
+- Docstrings include references to any scientific papers or standard publications on the underlying algorithm (e.g paper or Scikit-learn algorithm being implemented or a description of the algorithm used if nonstandard).
+
+
+## Review Checklist
+
+Aside from the general algorithm expectations outlined in the checklists above, code reviewers should use the following checklist to make sure the algorithm meets cuML standards. 
+
+### All
+
+- New files contain necessary license headers
+- Diff does not contain files with excess formatting changes, without other changes also being made to the file
+- Code does not contain any known serious memory leaks or garbage collection issues
+- Modifications are cohesive and in-scope for the PR's intended purpose
+- Changes to the public API will not have a negative impact to existing users between minor versions (eg. large changes to very popular public APIs go through a deprecation cycle to preserve backwards compatibility)
+- Where it is reasonable to do so, unexpected inputs fail gracefully and provide actionable feedback to the user
+- Automated tests properly exercise the changes in the PR
+- New algorithms provide benchmarks (both C++ and Python) 
```

Comment:
How about adding an item to see the list of printed slowest pytests in CI, to check that a new super slow test is not being introduced in the PR?